lib/sign.js

```diff
@@ -43,10 +50,9 @@ function sign (program, options, config) {
       console.error();
       return reject();
     }
-    cmd.validateProgram(program);
```

Comment:
this actually needs to happen within the promise so that an exception will reject the promise. You could accomplish it by returning a promise in signCmd and chaining them together, like this:
````
function signCmd (program, options, config) {
  return when.promise(function() {
      cmd.validateProgram(program);
    })
    .then(function() {
      return sign(_.assign(options, program), config);
    })
    .then(function (result) {
      logger.log(result.success ? 'SUCCESS' : 'FAIL');
      // ...
    });
}
````